src/backend/columnar/columnar_customscan.c

```diff
@@ -491,15 +534,401 @@ RelationIdGetNumberOfAttributes(Oid relationId)
 }
 
 
-static Path *
-CreateColumnarScanPath(PlannerInfo *root, RelOptInfo *rel, RangeTblEntry *rte)
+/*
+ * CheckVarStats() checks whether a qual involving this Var is likely to be
+ * useful based on the correlation stats. If so, or if stats are unavailable,
+ * return true; otherwise return false.
+ */
+static bool
+CheckVarStats(PlannerInfo *root, Var *var, Oid sortop)
+{
+	/*
+	 * Collect isunique, ndistinct, and varCorrelation.
+	 */
+	VariableStatData varStatData;
+	examine_variable(root, (Node *) var, var->varno, &varStatData);
+	if (varStatData.rel == NULL ||
+		!HeapTupleIsValid(varStatData.statsTuple))
+	{
+		return true;
+	}
+
+	AttStatsSlot sslot;
+	if (!get_attstatsslot(&sslot, varStatData.statsTuple,
+						  STATISTIC_KIND_CORRELATION, sortop,
+						  ATTSTATSSLOT_NUMBERS))
+	{
+		ReleaseVariableStats(varStatData);
+		return true;
+	}
```

Comment:
I got confused, shouldn't we return false for first two if's ?

Reply:
The comment at the top of the function says it returns "true" if we can't find the stats. Otherwise, we might not push down quals that are actually useful.